### PR TITLE
add click to view details of building on map for output widgets

### LINKED ISSUE #284: Displaying the value of the map features (damages and building attributes)
@Rashmil-1999 
Follow up things need to be done:
1. Move clicking event to the Map component; always assume the first layer is the layer (damage, functionality, dislocation layer) we read properties
2. Add a flag as props to turn on/off property selection clicking
3. Upon design ready:

```[tasklist]
### Tasks
  - [ ] https://github.com/IN-CORE/incore-playbook/issues/303
  - [ ] https://github.com/IN-CORE/incore-playbook/issues/304
```

-----

When clicking a building on map show the feature of the building such as damage, building attributes etc. It is needed especially when user compare two different scenarios. 

Technologies to consider:
- Query Geoserver WMS to return properties (which might be slow)
In order to display the property (attribute) of the building, we can use "WMS GetFeatureInfo" which supported by OpenLayers. Here is an example: https://openlayers.org/en/latest/examples/getfeatureinfo-layers.html
[9:22](https://ncsa-at-illinois.slack.com/archives/G9BH2ETSM/p1708010553157309)
another example: https://openlayers.org/en/latest/examples/getfeatureinfo-tile.html

~~Geojson (browser might not be able to handle all building in SLC using geojson). Need to rewrite openlayer part to work with geojson~~
~~vector tile (more performant version of geojson)~~

### LINKED ISSUE #303: Move feature table to side overlay with map

### PR DESCRIPTION
**Pull request by Rashmil-1999**
View the outputwidgets in step 3-4 for any widget that has a map with buildings on it, if you click any building it will show its detail below in a table.

### CODE REVIEW

**Comment by Rashmil-1999** [comment]
Fields that a building has:
- `guid`
- `year_built`
- `strctid`
- `struct_typ`
- `ffe_elev`
- `g_elev`
- `max_state`
- `sq_foot`
- `archetype`
- `a_stories`
- `appr_bldg`
- `appr_land`
- `appr_tot`
- `arch_flood`
- `arch_sw`
- `arch_wind`
- `b_stories`
- `broad_occ`
- `bsmt_type`
- `cont_val`
- `csv_sector`
- `csv_val_st`
- `dgn_lvl`
- `dwell_unit`
- `efacility`
- `gsq_foot`
- `major_occ`
- `no_stories`
- `nstra_cst`
- `nstrd_cst`
- `occ_detail`
- `occ_typ2`
- `occ_type`
- `parid`
- `repl_cst`
- `str_cst`
- `str_typ2`

I think the 1st 8-10 factors might be important but we can use this comment to strike out ones we don't need.

And, for Pole damage we have:

- `guid`
- `nodenwid`
- `epf_class`
- `max_state`
- `flow`
- `fltytype`
- `height`
- `spanLength`
- `strctype`
- `utilfcltyc`

File: src/components/childComponents/map/Map.jsx [hunk 1]
```diff
@@ -60,11 +63,55 @@ const Map = (props) => {
...
+				  evt.coordinate,
+				  viewResolution,
+				  viewProjection,
+				  {'INFO_FORMAT': 'application/json'}
+				);
+				if (url) {
+				  fetch(url)
+					.then((response) => response.json())
+					.then((json) => {
+						setSelectedBuilding(json.features[0].properties)
```
Comment (longshuicy) [inline src/components/childComponents/map/Map.jsx]: For some reason i have to close the table window then click the next building for the change to reflect. Could you check if that's the case?

Comment (Rashmil-1999) [reply src/components/childComponents/map/Map.jsx]: ohh, let me check that... might have something to do with state update...

Comment (Rashmil-1999) [reply src/components/childComponents/map/Map.jsx]: I don't think that there is any issue. It's just that the response takes time to load from geoserver. Let me see if could set some form of loading icon there to indicate that.

Comment (longshuicy) [reply src/components/childComponents/map/Map.jsx]: Ah i see. I guess i was too impatient haha. 
Yes please add a loading spinner (I think you can find an example in this repo as some of the other widget using loading spinner?)

File: src/components/childComponents/outputWidgets/BldgDamage.jsx [hunk 2]
```diff
@@ -144,15 +149,37 @@ export default function BldgDamage(props) {
 							zoom={mapConfig.ZOOM}
 							minZoom={mapConfig.MIN_ZOOM}
 							maxZoom={mapConfig.MAX_ZOOM}
-							extent={applyTransform(mapConfig.MAP_BOUNDS, getTransform("EPSG:4326", "EPSG:3857"))}
-							center={fromLonLat(mapConfig.MAP_CENTER)}
+							// extent={applyTransform(mapConfig.MAP_BOUNDS, getTransform("EPSG:4326", "EPSG:3857"))}
```
Comment (longshuicy) [inline src/components/childComponents/outputWidgets/BldgDamage.jsx]: If we are using 4326 could it be apply everywhere? 
Also could you remove the inline comments

**Review by longshuicy — CHANGES_REQUESTED** [review]
Everything works except i notice the building functionality widgets doesn't show the building layer anymore. Could you look into why?

image

**Comment by Rashmil-1999** [comment]
That only happens with try 9, try 8, try 6, try 4, try 3, try 2, Hurricane Ike 2, Datawolf run... basically, retrofit methods. Its there for Hurricane Ike and before... I don't think something here is causing issues... maybe from geoserver or datawolf?

**Comment by longshuicy** [comment]
> That only happens with try 9, try 8, try 6, try 4, try 3, try 2, Hurricane Ike 2, Datawolf run... basically, retrofit methods. Its there for Hurricane Ike and before... I don't think something here is causing issues... maybe from geoserver or datawolf?

I see. Let's get your PR merged in first then. We'll look into that later.

**Review by longshuicy — APPROVED** [review]

**Pull request merged**